gpii/node_modules/solutionsRegistry/src/js/SolutionsUtils.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * GPII Solutions Utils - Independent helper functions and utilities for working with
+ * solutions and their schemas.
+ *
+ * Copyright 2019 Raising the Floor - US
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013)
+ * under grant agreement no. 289016.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.universal.solutionsRegistry");
+
+/**
+ * This function will take a schema from a supported setting in the solutions
+ * registry and return metadata on it's default value, including the type of
+ * the default value. The function will always return a block of json in order
+ * to avoid any ambiguities for a situation where the default value is perhaps
+ * `null`. A key `hasDefault` will be `true` or `false` to communicate this.
+ * If the schema does have a default value, then the properties `type` and
+ * `default` can also be included. Note that there may not always be a `type`
+ * that can be included, the inclusion of this in the return if optional and
+ * will be included if a `type` is declared.
+ *
+ * This is NOT meant to be an exhaustive coverage of all the possible ways
+ * default values can be set for any JSON Schema. We merely want to cover the
+ * cases in use in the solutions registry, focussing on single primitive values,
+ * and situations where we next one level deep due to a requirement of having a
+ * `properties` entry in the schema.
+ *
+ * @param {Object} schema - Schema to search for a default value. This is what
+ * would be in the solutions registry under `supportedSettings` and a
+ * configuration. An example from the Windows Mouse Keys MaxSpeed setting is:
+ * ```json
+ * {
+ *     "title": "Mouse keys speed",
+ *     "description": "Speed of mouse keys",
+ *     "type": "number",
+ *     "multipleOf": 10,
+ *     "default": 80
+ * }
+ * ```
+ * @return {Object} Always returns a json block. For the example above the
+ * return value would be:
+ * ```json
+ * {
+ *     "hasDefault": true,
+ *     "type": "number",
+ *     "default": 80
+ * }
+ * ```
+ * If a default value cannot be detected the return payload will be:
+ * ```json
+ * {
+ *     "hasDefault": false
+ * }
+ * ```
+ */
+gpii.universal.solutionsRegistry.findDefaultValue = function (schema) {
```

Comment:
Well, this function doesn't seem to be used. Did I miss anything?

Reply:
This is used in other modules that are capturing settings and presenting them to the user, mostly the `gpii-app`.  But I'm also about to start using it in `gpii-devpmt`. It's in universal to be centrally located.

Reply:
It doesn't seem appropriate to use the universal as a central location to hold functions used by other repos but never been used by the universal itself.

@amb26, any suggestions?

Reply:
@cindyli Actually, this is part of GPII-4207 and I shouldn't have merged this in to this branch.

But, for the sake of this conversation, it seems to me like the universal repository would be the correct location for this.  It's a central utility of the solutions registry, and it seems highly likely that future code in universal would use it.